apps/web/src/actions/latte/partialRejectChanges.ts

```diff
@@ -0,0 +1,35 @@
+'use server'
+
+import { z } from 'zod'
+import { authProcedure } from '$/actions/procedures'
+import { partialDiscardLatteChanges as partialRejectLatteChanges } from '@latitude-data/core/services/copilot/latte/threads/checkpoints/partialRejectChanges'
+import { evaluateLatteThreadChanges } from '@latitude-data/core/services/copilot/latte/threads/evaluateChanges'
+
+export const partialRejectLatteChangesAction = authProcedure
+  .createServerAction()
+  .input(
+    z.object({
+      threadUuid: z.string(),
+      documentUuidsToReject: z.array(z.string()),
+    }),
+  )
+  .handler(async ({ ctx, input }) => {
+    const { workspace } = ctx
+    const { threadUuid, documentUuidsToReject } = input
+
+    await partialRejectLatteChanges({
+      workspace,
+      threadUuid,
+      documentUuidsToReject,
+    }).then((r) => r.unwrap())
+
+    // Evaluate the remaining changes (if any)
+    const result = await evaluateLatteThreadChanges({
+      threadUuid,
+      accepted: false, // Mark as rejected since we're rejecting specific changes
```

Comment:
This is contradictory to the partial acceptance. If the user accepts some of the changes is accepted for me. Does not matter if they partially reject some of them.

Reply:
Exactly. We don't know if they will reject all. So marking as rejected the eval is not ok imo. Only when they use the `undo` button or when is the last partial reject present